### PR TITLE
DR-818 Make create async; add profile and session setup

### PR DESCRIPTION
**Pull request by ddietterich**
Top-level changes:
1. Change the `checkApiUpToDate` task to use `svn` to pull the development version of the openApi yaml from the datarepo github.
2. Add a `updateApi` task to use `svn` to pull the development version of the openApi yaml and insert it into the CLI code.
3. Updated the README

Changes to main:
1. Change the create dataset calls to be async
2. Fix an NPE processing the `help` command when a command has no alternate names
3. Make a shared `--format` parser option
4. Add format option to the `profile show` command

Changes to test:
1. Add `session set basepath` so the CLI command and the http call are targeted at the same backend server.
2. Automate profile detection/creation; it was dependent on the dev environment

### CODE REVIEW

File: src/main/java/bio/terra/model/DRElement.java [hunk 1]
```diff
@@ -40,12 +40,14 @@ public boolean isLeaf() {
...
-        } else {
-            CommandUtils.printErrorAndExit("Invalid format; only text and json are supported");
+        switch (CLIFormatFlags.lookup(format)) {
+            case CLI_FORMAT_TEXT:
+                describeText();
+                break;
+
+            case CLI_FORMAT_JSON:
+                describeJson();
+                break;
```
Comment (marikomedlock) [inline src/main/java/bio/terra/model/DRElement.java:51]: I think we should have a default case in the switch that prints the error and exits.

File: src/test/java/bio/terra/endpoints/CLITestingConfig.java [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+        dataRepoURL = System.getenv("JADECLI_TEST_DATAREPO_URL");
+        if (dataRepoURL == null) {
+            dataRepoURL = "http://localhost:8080/";
+        }
+
+        // Secrets default is null; default is supplied in the main code
+        clientSecretsFilePath = System.getenv("JADECLI_TEST_CLIENT_SECRETS");
+
+        // Billing account is required
+        billingAccount = System.getenv("JADECLI_TEST_BILLING_ACCOUNT");
```
Comment (marikomedlock) [inline src/test/java/bio/terra/endpoints/CLITestingConfig.java:22]: This environment variable may need to be added to Travis.

File: README.md [hunk 3]
```diff
@@ -63,22 +69,11 @@ You can generate a distribution as tar or zip with the commands:
...
-`cp ./jade-data-repo/src/main/resources/data-repository-openapi.yaml ./src/main/resources/data-repository-openapi.yaml`
-8. Run the unit and integrated tests to make sure the changes haven't broken anything.
-`./gradlew :testCLIUnit --console=verbose --info`
-`./gradlew :testCLIIntegrated --console=verbose --info`
-9. Check for any changes and commit them if found `git status` `git commit -m "update Data Repo OpenAPI file copy"`
+### Keeping OpenAPI file in sync with the Data Repo API repository
+There are two gradle tasks that can help with keeping the OpenAPI file in sync:
+- `checkApiUpToDate` - pulls the OpenAPI file from the jade-data-repo develop branch and diffs it against 
+the OpenAPI file in the CLI. It returns "API is NOT up to date" if there is skew.
+- `updateAPI` - pulls the OpenAPI file from the jade-data-repo develop branch and overwrites the CLI copy.
```
Comment (marikomedlock) [inline README.md:77]: With this approach, I think we can remove the git submodule.

**Review by marikomedlock — APPROVED** [review]
LGTM. Thanks for taking care of the profile TODO in the tests.

File: scripts/checkApiUpToDate.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# the file and do the compare.
+svnimage=$(which svn)
+if [ -z "$svnimage" ]; then
+  echo "No svn so cannot retrieve the api yaml file"
+  exit 2
+fi
+
+tmpfile=/tmp/data-repository-openapi.yaml
+svn export --force --quiet \
+  https://github.com/DataBiosphere/jade-data-repo/trunk/src/main/resources/data-repository-openapi.yaml \
```
Comment (thathert) [inline scripts/checkApiUpToDate.sh]: You could `curl` this instead: https://raw.githubusercontent.com/DataBiosphere/jade-data-repo/develop/src/main/resources/data-repository-openapi.yaml

I think curl will be more likely to be on someone's system.

File: src/test/resources/CLICommandTests/outputDatasetCreate.txt [hunk 5]
```diff
@@ -2,7 +2,7 @@ class DatasetSummaryModel {
     id: %id%
     name: CLITestDataset
     description: This is a sample dataset definition
-    defaultProfileId: a003922d-a01f-4d95-8bc2-02719ceea1c5
+    defaultProfileId: %profileid%
```
Comment (thathert) [inline src/test/resources/CLICommandTests/outputDatasetCreate.txt]: nit: camelCase here?

**Review by thathert — APPROVED** [review]
LGTM, two small nits.

File: src/test/java/bio/terra/endpoints/CLITestingConfig.java (hunk 2 above)
Comment (thathert) [reply src/test/java/bio/terra/endpoints/CLITestingConfig.java:22]: Yep! Good catch. I updated it in the settings.

**Pull request merged**